### DISCUSSION TITLE
[Question] 2 copys of chunks on one chunkserver

### DISCUSSION THREAD

**Discussion by Sveeeeeen** [moosefs/moosefs#539] [Q&A]
Hello!

I have 2 chunk servers. One with around 20 TB  (RAID5) and another one with around 60 TB (JBOD).
My goal is set to 2 for all files. I just noticed moosefs can't create all 2 copies for all chunks. One chunkserver has much more space than the other. I get it.

But is it possible since the 2nd chunkserrver is using JBOD to create a 2nd copy on another drive? My goal is to have 2 copies of all chunks. No matter, it's on the same chunkserver (but better if not if space is available).

My mode is  already set to "loose". 

Any ideas?

**Comment by chogata** [comment]
You can't have 2 copies of the same chunk on one chunk server (process) - it's the way MooseFS is designed. A second copy will always be considered a duplicate. A solution would be to set up 2 processes of chunk server on the machine with JBOD, give each process half of the drives to keep chunks on. Label the chunk server on the machine with raid with one letter (for example R) and both the chunk servers on the machine with JBOD with another letter (for example J). Then define your storage class as `-K J,R` and in loose mode. MooseFS will try to keep one copy on J, one copy on R but when R becomes unavailable, it will keep both copies on J labeled chunk servers - one on each.

**Reply by Sveeeeeen** [reply]
I was thinking about the same and probably going for it. Maybe I missed something and there would be a “cleaner” way.
Thank you.
